lib/puppet/util/windows/service.rb

```diff
@@ -775,9 +782,14 @@ def wait_on_pending_state(service, pending_state)
         elapsed_time = 0
         last_checkpoint = -1
         loop do
-          status = query_status(service)
-          state = status[:dwCurrentState]
-
+          state = nil
+          checkpoint = nil
+          wait_hint = nil
+          query_status(service) do |status|
```

Comment:
Possible use-after-free for similar reasons per my comment in `transition_service_state`